client/my-sites/email/form/mailboxes/components/mailbox-form-wrapper/index.tsx

```diff
@@ -27,15 +27,20 @@ const MailboxFormWrapper = ( {
 		[ isRtl ? 'textInputPrefix' : 'textInputSuffix' ]: `\u200e@${ formFields.domain.value }\u202c`,
 	};
 
+	let renderPosition = 0;
+
 	const commonFieldProps = ( field: MailboxFormFieldBase< string > ) => {
+		++renderPosition;
+
 		return {
 			field,
 			onFieldValueChanged,
 			onRequestFieldValidation: () => mailbox.validateField( field.fieldName ),
+			renderPosition,
 		};
 	};
 
-	const UserFormFields = () => {
+	const renderUserFormFields = () => {
```

Comment:
When it became important to dynamically determine the position of the form elements, I realised that inner components are essentially functions and are executed after the currently executing component has been fully evaluated.

```jsx
<>
    <div> Words and text </div>
    <DirectComponent />
    { someFunctionThatRendersComponents() }
</>
```
For example, `<DirectComponent />` will be rendered/called in its own context AFTER the `someFunctionThatRendersComponents()` is called. 

This makes determining visual element positions difficult. 
The workaround is to call a `renderer function` because then the scope of that function evaluates in the right order.